.evergreen/run-tests-serde.sh

```diff
@@ -1,7 +0,0 @@
-#!/bin/sh
```

Comment:
Why was this file removed?

Reply:
In #279, I updated `run-tests.sh`, `run-tests-u2i.sh`, and `run-tests-decimal128.sh` to each run the serde tests to cover our feature flags, so this script wasn't actually being used anymore.